proto/tendermint/abci/types.proto

```diff
@@ -307,10 +309,10 @@ message EventAttribute {
 //
 // One usage is indexing transaction results.
 message TxResult {
-  int64             height = 1;
-  uint32            index  = 2;
-  bytes             tx     = 3;
-  ResponseDeliverTx result = 4 [(gogoproto.nullable) = false];
+  int64               height = 1;
+  uint32              index  = 2;
+  tendermint.types.Tx tx     = 3;
+  ResponseDeliverTx   result = 4 [(gogoproto.nullable) = false];
```

Comment:
It's not yet clear to me what the connection between Tx.Key and TxResult.result.tx_id is. Is there any yet? I think I'll need to checkout the branch locally for reviewing this properly.

Reply:
Will this be where the app can overwrite the key? And the Tx.key is what identifies the Tx when evicting? Is my understanding correct here?